pkgs/development/python-modules/deap/default.nix

```diff
@@ -9,19 +18,18 @@ buildPythonPackage rec {
     hash = "sha256-zAHemJLfp9G8mAPasoiS/q0XfwGCyB20c2CiQOrXeP8=";
   };
 
-  propagatedBuildInputs = [ numpy matplotlib ];
+  propagatedBuildInputs = [ matplotlib numpy pytest ];
 
-  nativeCheckInputs = [ nose ];
   checkPhase = ''
-    nosetests --verbosity=3
+    runHook preCheck
+    ${python3.interpreter} -m unittest
+    runHook postCheck
   '';
 
   meta = with lib; {
     description = "DEAP is a novel evolutionary computation framework for rapid prototyping and testing of ideas.";
     homepage = "https://github.com/DEAP/deap";
     license = licenses.lgpl3;
-    maintainers = with maintainers; [ psyanticy ];
+    maintainers = with maintainers; [ getpsyched psyanticy ];
```

Comment:
Please add yourself to maintainers in a separate commit, `python311Packages.deap: add getpsyched as maintainers` because it is not related to the fix.